tgui/packages/tgui/interfaces/SecureSafe.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * @file
+ * @copyright 2022
+ * @author blackrep (https://github.com/blackrep)
+ * @license MIT
+ */
+
+
+import { useBackend } from '../backend';
+import { Stack, Box, ProgressBar, Section, Flex, Button, Blink } from '../components';
+import { Window } from '../layouts';
+
+import { capitalize, glitch } from './common/stringUtils';
+
+const KEY_PAD_INPUT_LAYOUT = [
+  ['7', '8', '9', 'A'],
+  ['4', '5', '6', 'B'],
+  ['1', '2', '3', 'C'],
+  ['0', 'F', 'E', 'D'],
+];
+
+const stylizeCode = (attempt, codeLen) => {
+  return attempt.padEnd(codeLen, "*").split("").join(" ");
+};
+
+const SecureSafeScreen = (props, _context) => {
+  const {
+    attempt,
+    codeLen,
+    disabled,
+    emagged,
+    padMsg,
+  } = props;
+
+  let content = padMsg ? padMsg : stylizeCode(attempt, codeLen);
+  if (disabled) {
+    content = "NO ACCESS";
+  }
+  if (emagged) {
+    content = glitch(content, 2);
+  }
+  return (
+    // Crude hack to get the styling I want
+    <ProgressBar
```

Comment:
We shouldn't be abusing the progress bar component like this. It is fine to add a bit of custom CSS for your interface for whatever purpose this is doing.